File: 算法/排序/简单选择排序.py
def choose_sort(unordered_ls: list) -> None:
    for i in range(len(unordered_ls) - 1):
        min = i
        for j in range(i, len(unordered_ls)):
            if unordered_ls[j] < unordered_ls[min]:
                min = j
        if min != i:
            unordered_ls[min], unordered_ls[i] = unordered_ls[i], unordered_ls[min]


if __name__ == "__main__":
    ls = [20, 30, 10, 40, 50]
    choose_sort(ls)

    for i in ls:
        print(i, end=' ')

File: 算法/排序/简单选择排序.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

template<class T>
static void choose_sort(vector<T> &v) {
    for (int i = 0; i < v.size() - 1; i++) {
        int min = i;
        for (int j = i; j < v.size(); j++)
            if (v[min] > v[j]) min = j;
        if (min != i) swap(v[min], v[i]);
    }
}

int main() {
    vector<int> v = {3, 1, 2, 4, 5};

    choose_sort(v);

    for (const auto &var: v)
        cout << var << " ";
    cout << endl;

    return 0;
}
